src/xSTIR/pSTIR/STIR.py

```diff
@@ -1625,6 +1625,31 @@ def get_time_at_which_num_prompts_exceeds_threshold(self, threshold):
         return v
 
 
+class PoissonNoiseGenerator(object):
+    """
+    Class that generates Poisson noise.
+    """
+
+    def __init__(self, scaling_factor=1.0, preserve_mean=False):
+        self.name = "PoissonNoiseGenerator"
+        self.handle = pystir.cSTIR_createPoissonNoiseGenerator(scaling_factor, preserve_mean)
+        check_status(self.handle)
+        self.scale = scaling_factor
+
+    @property
+    def scaling_factor(self):
+        return self.scale
```

Comment:
if we have `preserve_mean` in the constructor, shouldn't we have the property?

Reply:
My python isn't good enough, but is this a settable property? If so, it's dangerous as we don't update the STIR object